tests/testcore/functional_test_base.go

```diff
@@ -193,7 +201,17 @@ func (s *FunctionalTestBase) SetupSuiteWithCluster(clusterConfigFile string, opt
 
 	// Logger might be already set by the test suite.
 	if s.Logger == nil {
-		s.Logger = log.NewTestLogger()
+		tl := testlogger.NewTestLogger(s.T())
+
+		// Instead of failing immediately, TearDownTest will check for unexpected
+		// errors after each test completed. This is better since otherwise is would fail inside
+		// the server and not the test, creating a lot of noise and possibly stuck tests.
+		testlogger.DontFailOnError(tl)
+
+		// Fail test when an assertion fails (see `softassert` package).
+		tl.DontExpect(testlogger.Error, ".*", tag.FailedAssertion())
```

Comment:
At first I tried to explicitly `Expect` all errors, but after number 20 I realized that was a foolish endeavour.